### PR TITLE
Deal and Product mapping

### PR DESCRIPTION
**Pull request by mehreen-tahir**
Entity mapping and rspec test cases added for Deal.
Entity mapping and rspec test cases added for Product.

### CODE REVIEW

File: app/models/entities/deal.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class Entities::Deal < Maestrano::Connector::Rails::Entity
+  def self.connec_entity_name
+    'Opportunities'
```
Comment (MarcoBgn) [inline app/models/entities/deal.rb]: You can use 'Opportunity' here, will be pluralised by the framework

File: app/models/entities/product.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  def self.object_name_from_external_entity_hash(entity)
+    entity['name']
+  end
+end
+
+class ProductMapper
+  extend HashMapper
+
+  map from('name'), to('name')
```
Comment (MarcoBgn) [inline app/models/entities/product.rb:26]: Is there any information about quantity or availability?

**Review by MarcoBgn — COMMENTED** [review]
Thanks for the PR

File: app/models/entities/deal.rb (hunk 1 above)
Comment (mehreen-tahir) [reply app/models/entities/deal.rb]: Thank you, feedback incorporated.

File: app/models/entities/product.rb (hunk 2 above)
Comment (mehreen-tahir) [reply app/models/entities/product.rb:26]: Feedback incorporated for adding availability to product.

File: app/models/maestrano/connector/rails/external.rb [hunk 3]
```diff
@@ -15,6 +15,6 @@ def self.create_account_link(organization = nil)
   end
 
   def self.entities_list
-    %w(organization)
+    %w(organization deal product)
```
Comment (MarcoBgn) [inline app/models/maestrano/connector/rails/external.rb:18]: Just as a remainder for the next PR, the order here is important for references. Please add 'app_users' at the beginning of the list, and put product before deal. 
```ruby
%w(app_user organization product deal)
```

Comment (mehreen-tahir) [reply app/models/maestrano/connector/rails/external.rb:18]: Sure. I will keep it in reference.

**Pull request merged**